src/QsCompiler/Core/SyntaxTreeTransformation.fs

```diff
@@ -40,36 +42,44 @@ type QsSyntaxTreeTransformation<'T> private (state : 'T, unsafe : string) =
 
     member this.InternalState = state
 
-    new (state) as this =
-        QsSyntaxTreeTransformation(state, "unsafe") then
-            this.Types           <- new TypeTransformation<'T>(this)
-            this.ExpressionKinds <- new ExpressionKindTransformation<'T>(this)
-            this.Expressions     <- new ExpressionTransformation<'T>(this)
-            this.StatementKinds  <- new StatementKindTransformation<'T>(this)
-            this.Statements      <- new StatementTransformation<'T>(this)
-            this.Namespaces      <- new NamespaceTransformation<'T>(this)
+    new (state : 'T, options : TransformationOptions) as this =
+        QsSyntaxTreeTransformation<'T>(state, "unsafe") then
+            this.Types           <- new TypeTransformation<'T>(this, options)
+            this.ExpressionKinds <- new ExpressionKindTransformation<'T>(this, options)
+            this.Expressions     <- new ExpressionTransformation<'T>(this, options)
+            this.StatementKinds  <- new StatementKindTransformation<'T>(this, options)
+            this.Statements      <- new StatementTransformation<'T>(this, options)
+            this.Namespaces      <- new NamespaceTransformation<'T>(this, options)
+
+    new (state : 'T) = new QsSyntaxTreeTransformation<'T>(state, TransformationOptions.Default)
 
 
-and TypeTransformation<'T> internal (unsafe) =
-    inherit TypeTransformationBase()
+and TypeTransformation<'T> internal (options, unsafe) =
+    inherit TypeTransformationBase(options)
     let mutable _Transformation : QsSyntaxTreeTransformation<'T> option = None
 
     member this.Transformation
```

Comment:
I wonder if it would also be possible to have the property be called `State`, so that transformations can access their state using `this.State.Foo`? Since the state seems like it will be referenced a lot by transformations (if it's a stateful transformation), I think it would be good for it to be short, unless there's some reason to be explicit about internal state vs. other state.

Reply:
Adding the shortcut makes sense. I might have gone with something like SharedState then (which is a bit longer) to distinguish it form any state within the subtransformation only. Thoughts?

Reply:
I think `SharedState` makes sense.